### ISSUE TITLE
Inaccuracy in file remove_duplicates_list.py

### ISSUE THREAD

**Issue by Mahyar-GH** [MakeContributions/DSA#1249]
**Details**
I'm pretty sure [remove_duplicates_list.p](https://github.com/MakeContributions/DSA/blob/main/algorithms/Python/arrays/remove_duplicates_list.py#L3) is not O(n) time complexity.
Because of this line:
https://github.com/MakeContributions/DSA/blob/d3c2184af831d7e227931b299df6775da7867b18/algorithms/Python/arrays/remove_duplicates_list.py#L25

**Comment by noor-malaika** [comment]
I agree that the time cmplx is not O(n) but i think the reason is not the comparison statement.

The line of _**list.remove(value)**_ looks for the value in list while it is nested in a loop of n iters, so in worst case it will have a complexity of O(n^2).

**Comment by Mahyar-GH** [comment]
> I agree that the time cmplx is not O(n) but i think the reason is not the comparison statement.
> 
> The line of _**list.remove(value)**_ looks for the value in list while it is nested in a loop of n iters, so in worst case it will have a complexity of O(n^2).

To my knowledge using the in keyword followed by a python list is O(n).
If it was followed a set though, it would have been constant time.

Anyways you are also correct the line below is itself O(n). I haven't paid attention to that.

**Comment by vishalkhyatriya12** [comment]
Hi this is vishal!!
lets make it more simpler 
here is the code
sample_case = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]

def make_distinct(values: list) -> list:
    """
    Remove duplicate elements in an array inplace without creating new array.
    """
    index = len(values) - 1
    while index > 0:
        if values[index] in values[:index]:
            values.pop(index)
        index -= 1

    return values

if __name__ == "__main__":
    print(make_distinct(sample_case))
    
use while instead of for loop and finally o(n)
